tuned/plugins/base.py

```diff
@@ -153,15 +153,19 @@ def _get_device_objects(self, devices):
 		return None
 
 	def _get_matching_devices(self, instance, devices):
-		if instance.devices_udev_regex is None:
-			return set(self._device_matcher.match_list(instance.devices_expression, devices))
+		if (instance.cpuinfo_regex is None or re.search(self._variables.expand(instance.cpuinfo_regex), self._cmd.read_file("/proc/cpuinfo"), re.MULTILINE)) and \
+			(instance.uname_regex is None or re.search(self._variables.expand(instance.uname_regex), " ".join(os.uname()))):
```

Comment:
I think this is really abusing the `_get_matching_devices` method. Ideally, this would be handled some place else. Not sure where at the moment.

Reply:
We could split it to two methods, but I am not sure whether it helps.